bumble/device.py

```diff
@@ -2110,11 +2110,29 @@ async def update_connection_parameters(
         supervision_timeout,
         min_ce_length=0,
         max_ce_length=0,
-    ):
+        use_l2cap=False,
+    ) -> bool:
         '''
         NOTE: the name of the parameters may look odd, but it just follows the names
         used in the Bluetooth spec.
         '''
+
+        if use_l2cap:
+            if connection.role != BT_PERIPHERAL_ROLE:
+                raise InvalidStateError(
+                    'only peripheral can update connection parameters with l2cap'
+                )
+            l2cap_result = (
+                await self.l2cap_channel_manager.update_connection_parameters(
+                    connection,
+                    connection_interval_min,
+                    connection_interval_max,
+                    max_latency,
+                    supervision_timeout,
+                )
+            )
+            return l2cap_result == 0
```

Comment:
This should raise an error (`HCI_StatusError(l2cap_result)`) if the result is not HCI_SUCCESS, just like the non-l2cap version below.

Reply:
Now raising `ConnectionParameterUpdateError` defined in core.py